attributes/default.rb

```diff
@@ -238,26 +241,55 @@
   'default' => "659400f9842fffb8d64ae0b650f081b9",
   'ubuntu' => { 'default' => "c2b410031867dc7c966ca5b1aa0c72aa" }
 )
+
+# MySQL Packages
+default['cluster']['mysql']['package']['root'] = "https://#{node['cluster']['region']}-aws-parallelcluster.s3.#{node['cluster']['region']}.#{node['cluster']['aws_domain']}/archives/mysql"
+default['cluster']['mysql']['package']['version'] = "8.0.31-1"
+default['cluster']['mysql']['package']['source-version'] = "8.0.31"
+default['cluster']['mysql']['package']['platform'] = if arm_instance?
+                                                       value_for_platform(
+                                                         'default' => "el/7/aarch64"
+                                                       )
+                                                     else
+                                                       value_for_platform(
+                                                         'default' => "el/7/x86_64",
+                                                         'ubuntu' => {
+                                                           '20.04' => "ubuntu/20.04/x86_64",
+                                                           '18.04' => "ubuntu/18.04/x86_64",
+                                                         }
+                                                       )
+                                                     end
+default['cluster']['mysql']['package']['file-name'] = "mysql-community-client-#{node['cluster']['mysql']['package']['version']}.tar.gz"
+default['cluster']['mysql']['package']['archive'] = "#{node['cluster']['mysql']['package']['root']}/#{node['cluster']['mysql']['package']['platform']}/#{node['cluster']['mysql']['package']['file-name']}"
+default['cluster']['mysql']['package']['source'] = "#{node['cluster']['mysql']['package']['root']}/source/mysql-#{node['cluster']['mysql']['package']['source-version']}.tar.gz"
+
+# MySQL Validation
 if arm_instance?
   default['cluster']['mysql']['repository']['packages'] = value_for_platform(
-    'default' => %w(mysql-community-devel mysql-community-libs mysql-community-common mysql-community-client-plugins),
+    'default' => %w(mysql-community-devel mysql-community-libs mysql-community-common mysql-community-client-plugins mysql-community-libs-compat),
     'ubuntu' => {
-      'default' => %w(libmysqlclient-dev libmysqlclient21 mysql-common),
```

Comment:
Because, for Ubuntu (18|20).04 Arm, the version of mysql-common is not aligned withe the versions of the other two packages (For Ubuntu Arm, the packages don't come from MySQL, but from an ec2 repo). What would be a better way to account for the version differences in this case?

Reply:
So now we are taking `mysql-common` package from our artifacts on S3?
I do not have the whole context on the activity, so I may be wrong: what looks a bit fragile to me is taking packages that require to be aligned part from repos and part from our S3 bucket. If this is the case, isn't this approach exposed to packages misalignment in the future?

Reply:
Point clarified offline by @jdeamicis.